libexec/rtld-elf/aarch64/rtld_start.S

```diff
@@ -315,6 +315,21 @@ ENTRY(_rtld_tlsdesc_dynamic)
 	 *
 	 */
 1:
+#ifdef RTLD_SANDBOX
+	/*
+	 * Caller must be Restricted, so save its rcsp at the bottom of its own
```

Comment:
Trusted only means that the callee runs on the stack of the caller, which is restricted most of the time.

(Although I think none of the functions I listed in `trusted_globals_names` actually uses TLS or locales. They are supposed to 'pure' functions that do not access any globals.)

Reply:
* strcasestr uses the locale (how else does it know what case-insensitive means?)
* vfork and rfork make a syscall that can fail and thus set errno (calls cerror, which gets the pointer from __error), though it's true that uses `&_get_curthread()->error` instead

Reply:
* Good catch. I somehow assumed that `strcasestr` only works for ASCII.
* vfork and rfork are in the list not because they are 'trusted' per se, but because they simply won't work if trampolines are used when calling them. So this is really a hack.
* __libc_start1 is in the list because it calls `main` via a function pointer, which currently does not cause a domain transition. So this is also a hack.